app/client/service_runner.py

```diff
@@ -88,7 +85,7 @@ def is_queue_empty():
             raise Exception('Master service did not become idle before timeout.')
 
     def kill(self):
-        self._run_service('{} stop'.format(self._main_executable))
+        self._run_service([self._main_executable, 'stop'])
 
     def _run_service(self, cmd, service_url=None):
```

Comment:
I'm not familiar with this Popen_with_delayed_expansion(), but I'm assuming that you can interchange a string with a list of strings?